### PR TITLE
alvr: init at 20.6.1

### PR DESCRIPTION
**Pull request by PassiveLemon**
## Description of changes
Adds ALVR
https://github.com/NixOS/nixpkgs/issues/238102

Packaged with the prebuilt app-image because building this from source is painful. If you get a source build working and want me to add it, feel free to comment.

I goofed up and accidentally mass pinged in https://github.com/NixOS/nixpkgs/pull/275926
I apologize for anyone involved

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Lurkki14** [comment]
The changes to `nixos/` should be under a separate commit

File: nixos/modules/programs/alvr.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+with lib;
+
+let
+  cfg = config.programs.alvr;
+in
+{
+  options = {
+    programs.alvr = {
+      enable = mkEnableOption (lib.mdDoc "ALVR");
```
Comment (Lurkki14) [inline nixos/modules/programs/alvr.nix:11]: You should state in the enable option what ALVR is

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

**Comment by Lurkki14** [comment]
An entry for new modules should be added to `nixos/doc/manual/release-notes/rl-2405.section.md`

**Comment by PassiveLemon** [comment]
Updated with requested changes

File: nixos/modules/programs/alvr.nix (hunk 1 above)
Comment (PassiveLemon) [reply nixos/modules/programs/alvr.nix]: Done

File: nixos/modules/programs/alvr.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+with lib;
+
+let
+  cfg = config.programs.alvr;
+in
+{
+  options = {
+    programs.alvr = {
+      enable = mkEnableOption (lib.mdDoc "Whether to enable ALVR, the VR desktop streamer.");
```
Comment (Lurkki14) [inline nixos/modules/programs/alvr.nix:11]:
```suggestion
      enable = mkEnableOption (lib.mdDoc "ALVR, the VR desktop streamer");
```
This is added by `mkEnableOption` already

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: nixos/modules/programs/alvr.nix (hunk 2 above)
Comment (PassiveLemon) [reply nixos/modules/programs/alvr.nix]: done

**Review by Lurkki14 — APPROVED** [review]

File: pkgs/by-name/al/alvr/package.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+in
+appimageTools.wrapType2 {
+  inherit pname version src;
+
+  extraInstallCommands = ''
+    mv $out/bin/${pname}-${version} $out/bin/${pname}
+
+    install -Dm444 ${appimageContents}/alvr.desktop -t $out/share/applications
+    substituteInPlace $out/share/applications/alvr.desktop \
+      --replace 'Exec=alvr_dashboard' 'Exec=${pname}'
```
Comment (h7x4) [inline pkgs/by-name/al/alvr/package.nix:23]: I don't think the executable name should depend on the package name. If anything, it depends on `meta.mainProgram`, but a string literal is probably fine.

```suggestion
    mv $out/bin/alvr-${version} $out/bin/alvr

    install -Dm444 ${appimageContents}/alvr.desktop -t $out/share/applications
    substituteInPlace $out/share/applications/alvr.desktop \
      --replace 'Exec=alvr_dashboard' 'Exec=alvr'
```

Comment (PassiveLemon) [reply pkgs/by-name/al/alvr/package.nix]: Seems reasonable, updated

**Comment by raspher** [comment]
Result of `nixpkgs-review pr 284154` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>

**Review by h7x4 — APPROVED** [review]

**Pull request merged**